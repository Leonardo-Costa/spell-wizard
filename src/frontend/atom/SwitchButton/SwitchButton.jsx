import { StyleSheet, Switch, View } from "react-native";
import React, { useState } from "react";
import colors from '../../misc/Colors';

function SwitchButton() {
  const [isEnable, setIsEnabled] = useState(false);
  const changeSwitch = () => {
    setIsEnabled(previousState => !previousState);
  }

  return (
    <View>
      <Switch trackColor={{
        false: colors.dark_gray,
        true: colors.light_gray
        }}
        thumbColor={
          !isEnable ? colors.light_gray : colors.dark_gray
        }
        onValueChange={changeSwitch}
        value={isEnable}
        style={styles.container}>
      </Switch>
  </View>
  )
  
}

const styles = StyleSheet.create({
  container: {
    margin: 5,
  }
});

export { SwitchButton }