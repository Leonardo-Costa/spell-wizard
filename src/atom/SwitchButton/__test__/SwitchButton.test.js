import React from 'react';
import renderer from 'react-test-renderer';

import {SwitchButton} from '../SwitchButton';

it('renders correctly', () => {
    const tree = renderer.create(<SwitchButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });