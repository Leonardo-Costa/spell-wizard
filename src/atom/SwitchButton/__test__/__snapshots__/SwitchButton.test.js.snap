// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#D6D6D6"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "margin": 5,
        },
      ]
    }
    thumbTintColor="#D6D6D6"
    tintColor="#202020"
    value={false}
  />
</View>
`;
